L1Trigger/L1THGCal/plugins/ntuples/HGCalTriggerNtupleHGCDigis.cc

```diff
@@ -212,22 +157,68 @@ fill(const edm::Event& e, const edm::EventSetup& es)
         hgcdigi_wafer_.emplace_back(id.wafer());
         hgcdigi_wafertype_.emplace_back(id.waferType());
         hgcdigi_cell_.emplace_back(id.cell());
-        hgcdigi_cell_eta_.emplace_back(geom_fh->getPosition(id.rawId()).eta());
-        hgcdigi_cell_phi_.emplace_back(geom_fh->getPosition(id.rawId()).phi());
-        hgcdigi_cell_z_.emplace_back(geom_fh->getPosition(id.rawId()).z());
+        GlobalPoint cellpos = geom_fh->getPosition(id.rawId());
+        hgcdigi_eta_.emplace_back(cellpos.eta());
+        hgcdigi_phi_.emplace_back(cellpos.phi());
+        hgcdigi_z_.emplace_back(cellpos.z());
         hgcdigi_data_.emplace_back(digi[2].data()); 
-        int is_tot=0,is_adc=0;
-        if (digi[2].mode()) is_tot=1;
-        else is_adc =1;
-        hgcdigi_istot_.emplace_back(is_tot);
+        int is_adc=0;
+        if (!(digi[2].mode())) is_adc =1;
         hgcdigi_isadc_.emplace_back(is_adc);
         if (is_Simhit_comp_) {
           double hit_energy=0;
-          auto itr = simhits_fh.find(id);
-          if(itr!=simhits_fh.end())hit_energy = itr->second;
-          hgcdigi_cell_simhit_energy_.emplace_back(hit_energy); 
+          auto itr = simhits_fh_.find(id);
+          if(itr!=simhits_fh_.end())hit_energy = itr->second;
+          hgcdigi_simenergy_.emplace_back(hit_energy); 
+        }
+      }
+}
+
+void
+HGCalTriggerNtupleHGCDigis::
+simhits(const edm::Event& e)
+{
+
+      edm::Handle<edm::PCaloHitContainer> ee_simhits_h;
+      e.getByToken(SimHits_inputee_,ee_simhits_h);
+      const edm::PCaloHitContainer& ee_simhits = *ee_simhits_h;
+      edm::Handle<edm::PCaloHitContainer> fh_simhits_h;
+      e.getByToken(SimHits_inputfh_,fh_simhits_h);
+      const edm::PCaloHitContainer& fh_simhits = *fh_simhits_h;
+      
+      //EE
+      int layer=0,cell=0, sec=0, subsec=0, zp=0,subdet=0;
+      ForwardSubdetector mysubdet;
+      HGCalDetId recoDetId ;
+      
+      for( const auto& simhit : ee_simhits ) { 
+        HGCalTestNumbering::unpackHexagonIndex((HGCalDetId)simhit.id(), subdet, zp, layer, sec, subsec, cell); 
```

Comment:
Is it needed to cast to HGCalDetId? 
simhit.id() returns an unsigned int and unpackHexagonIndex takes an unsigned int as input.